### PR TITLE
Add Decimal128 support

### PR DESCRIPTION
**Pull request by serjshevchenko**
Fix Issue #161

### CODE REVIEW

**Comment by serjshevchenko** [comment]
As mentioned in Issue #161  descr., this only works since mongo3.4 cause only then Dicimal128 added.

File: umongo/fields.py [hunk 1]
```diff
@@ -152,7 +152,12 @@ class IntegerField(BaseField, ma_fields.Integer):
 
 
 class DecimalField(BaseField, ma_fields.Decimal):
-    pass
```
Comment (lafrech) [inline umongo/fields.py]: Please add a docstring with at least a warning about MongoDB version support (3.4+).

File: umongo/fields.py [hunk 2]
```diff
@@ -152,7 +152,12 @@ class IntegerField(BaseField, ma_fields.Integer):
...
+    def _serialize_to_mongo(self, obj):
+        return Decimal128(obj)
+
+    def _deserialize_from_mongo(self, value):
+        if value:
+            return value.to_decimal()
```
Comment (lafrech) [inline umongo/fields.py]: There should be an explicit else.

And what if the value is decimal `0`?

Comment (lafrech) [reply umongo/fields.py]: Or perhaps it should just read

```py
    def _deserialize_from_mongo(self, value):
        return value.to_decimal()
```

File: tests/test_fields.py [hunk 3]
```diff
@@ -570,3 +572,53 @@ class MyDoc(Document):
...
+
+        class MySchema(Schema):
+            price = fields.DecimalField(attribute='in_mongo_price')
+
+        MyDataProxy = data_proxy_factory('My', MySchema())
+        d = MyDataProxy()
+        d.load({'price': Decimal128('12.5678')})
+        assert d.dump() == {'price': Decimal('12.5678')}
+        assert d.to_mongo() == {'in_mongo_price': Decimal128('12.5678')}
+
```
Comment (lafrech) [inline tests/test_fields.py]: Missing

        assert d.get('price') == Decimal('12.5678')

?

**Comment by lafrech** [comment]
I added the mashmallow<3 constraint in another commit for clarity.

@serjshevchenko, please

- rebase (remove the setup.py modification from your commit)
- add yourself to AUTHORS.rst
- add missing line in the tests (see my comment above)

And I think this is good to go.

AFAIU, this is not a breaking change as the Decimal was already broken on MongoDB < 3.4. I'm afraid the error for people using MongoDB<3.4 won't be very helpful. When trying to record a Decimal128 into a MongoDB 3.2 instance, I get:

    raise AutoReconnect("connection closed")
    pymongo.errors.AutoReconnect: connection closed

I have no idea how to do better.

@touilleMan, thoughts ? Any objection to releasing this as 1.2.0?

**Comment by serjshevchenko** [comment]
after rebasing, coverage decreased.
Code which I added, is coverage fully.

**Comment by lafrech** [comment]
Don't mind the coverage too much. It sometimes returns weird results. Covering your own change is enough.

@touilleMan, do you want to take a look before we push a 1.2.0 with this?

**Comment by lafrech** [comment]
Alright. What's done can be undone. I took it upon myself to merge this and https://github.com/Scille/umongo/pull/165.

@touilleMan, I plan to release 1.2.0 by the end of the week.

**Comment by touilleMan** [comment]
@lafrech sorry for the lack of reply (I spend a month in the Pilippines with barely no internet)
I have no objection on releasing 1.2.0. More generally, you're the most involved in this project currently and I trust your judgement so yo should feel free to release whenever you want ;-)

**Pull request merged**